### PR TITLE
Revives: Plates, Bowls and More Plates

### PR DESCRIPTION
**Pull request by Alberyk**
This pr brings back this pr #13269

### CODE REVIEW

**Review by alsoandanswer — APPROVED** [review]

File: code/modules/cooking/machinery/cooking_machines/container.dm [hunk 1]
```diff
@@ -249,10 +249,10 @@
 	return TRUE
 
 /obj/item/reagent_containers/cooking_container/plate
```
Comment (Arrow768) [inline code/modules/cooking/machinery/cooking_machines/container.dm:251]: You should update the path to match the new name of the item.

**Review by Arrow768 — DISMISSED** [review]

File: code/modules/cooking/trays.dm [hunk 2]
```diff
@@ -206,3 +206,15 @@
 	if(istype(A,/obj/structure/table))
 		safedrop = TRUE
 	return ..(A, user, click_parameters)
+
+/obj/item/tray/plate
+	name = "serving plate"
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "l_plate"
+	desc = "A large plate for serving meals on."
```
Comment (SleepyGemmy) [inline code/modules/cooking/trays.dm:214]:
```suggestion
	desc = "A large plate for serving meals on."
	icon = 'icons/obj/kitchen.dmi'
	icon_state = "l_plate"
```

File: code/modules/cooking/trays.dm [hunk 3]
```diff
@@ -206,3 +206,15 @@
...
+	throwforce = 4
+	force = 3
+	flags = null
+	matter = list(DEFAULT_TABLE_MATERIAL = 1000)
+	recyclable = TRUE
+	max_carry = 7 //That's 3 dishes, a knife, spoon and fork and a glass
```
Comment (SleepyGemmy) [inline code/modules/cooking/trays.dm:220]:
```suggestion
	max_carry = 7 // That's 3 dishes, a knife, spoon and fork and a glass
```

**Review by SleepyGemmy — DISMISSED** [review]

File: code/modules/cooking/plates.dm [hunk 4]
```diff
@@ -0,0 +1,173 @@
+/*
+Plates that can hold your cooking stuff
+*/
+//Click with food to put on plate
+//Click with cutlery to take some
+//Click in hand to remove food
```
[the remark this answers is not in the document]
Comment (Alberyk) [reply code/modules/cooking/plates.dm:6]: Done.

File: code/modules/cooking/plates.dm [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+				var/image/I = new(U.icon, "loadedfood")
+				I.color = reagents.get_color()
+				U.add_overlay(I)
+				reagents.trans_to_obj(U, min(reagents.total_volume,U.transfer_amt))
+				U.is_liquid = TRUE
+				return
+
+/obj/item/reagent_containers/bowl/on_reagent_change()
+	if(!reagents.total_volume && !grease)
+		grease = TRUE
```
[the remark this answers is not in the document]
Comment (Alberyk) [reply code/modules/cooking/plates.dm:56]: Done.

File: code/modules/cooking/machinery/cooking_machines/container.dm [hunk 6]
```diff
@@ -301,7 +301,21 @@
...
+	if(length(contents)) // Only if something was actually added.
+		icon_state = "[initial(icon_state)]_prep"
+
+/obj/item/reagent_containers/cooking_container/board/on_reagent_change()
+	. = ..()
+	if(reagents.total_volume) // Only if something was actually added.
+		icon_state = "[initial(icon_state)]_prep"
+
+/obj/item/reagent_containers/cooking_container/board/bowl
 	name = "serving bowl"
```
Comment (SleepyGemmy) [inline code/modules/cooking/machinery/cooking_machines/container.dm:319]:
```suggestion
	name = "mixing bowl"
```

File: code/modules/cooking/plates.dm [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+/obj/item/reagent_containers/bowl
+	name = "bowl"
+	desc = "A small bowl for serving liquid meals in."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "bowl"
+	fragile = 3
+	shatter_material = DEFAULT_TABLE_MATERIAL // Slight typecasting abuse here, gets converted to a material in Initialize().
+	can_be_placed_into = list()
+	flags = OPENCONTAINER
+	var/grease = FALSE
```
Comment (SleepyGemmy) [inline code/modules/cooking/plates.dm:17]: is this set to FALSE when put in a utility vendor?

File: code/modules/cooking/plates.dm [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+		LAZYADD(O, I)
+	set_overlays(O)
+	return ..()
+
+/obj/item/reagent_containers/bowl/plate
+	name = "plate"
+	desc = "A plate for dishing up the finest of cuisine."
+	desc_info = "Click with food to put food on.<br>\
+		- Click with cutlery to eat some.<br>\
+		- Click it with the active hand to remove food"
```
Comment (SleepyGemmy) [inline code/modules/cooking/plates.dm:102]:
```suggestion
```

File: code/modules/cooking/plates.dm [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+Plates that can hold your cooking stuff
+*/
+//Click with food to put on plate
+//Click with cutlery to take some
+//Click in hand to remove food
+
+/obj/item/reagent_containers/bowl
+	name = "bowl"
+	desc = "A small bowl for serving liquid meals in."
+	icon = 'icons/obj/kitchen.dmi'
```
Comment (SleepyGemmy) [inline code/modules/cooking/plates.dm:11]:
```suggestion
	desc = "A small bowl for serving liquid meals in."
	desc_info = "Click with food to put food on.<br>\
	- Click with cutlery to eat some.<br>\
	- Click it with the active hand to remove food."
	icon = 'icons/obj/kitchen.dmi'
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

File: code/modules/cooking/plates.dm [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+	else
+		return ..()
+	update_icon()
+	return
+
+/obj/item/reagent_containers/bowl/on_rag_wipe(obj/item/reagent_containers/glass/rag/R)
+	. = ..()
+	if(grease)
+		grease = FALSE
+	return
```
Comment (NonQueueingMatt) [inline code/modules/cooking/plates.dm:84]: Pretty sure this ought to update_icon() too.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/cooking/plates.dm (hunk 7 above)
Comment (Alberyk) [reply code/modules/cooking/plates.dm:20]: Yes

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**